### PR TITLE
Pin numpy version < 2.0

### PR DESCRIPTION
**Pull request by ErlendHaa**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,7 +95,7 @@ curve mnemonics as field names (column names).
 
 To develop dlisio, or to build a particular revision from source, you need:
 
-* A C++11 compatible compiler (tested daily on gcc, clang, and msvc 2015)
+* A C++11 compatible compiler (tested on gcc, clang, and msvc 2015)
```
Comment (achaikou) [inline README.md:98]: I think it is 2019 now. Should we update it or remove it?

File: python/docs/index.rst [hunk 2]
```diff
@@ -1,6 +1,6 @@
-.. image:: ../../dlisio-logo.svg
+.. image:: ../../dlisio-icon.svg
    :width: 400
-   :alt: dlisio logo
+   :alt: dlisio icon
    :align: center
```
Comment (achaikou) [inline python/docs/index.rst:5]: See two more "logo" words in tests. Shouldn't affect anything, but change?

File: python/requirements-dev.txt [hunk 3]
```diff
@@ -1,6 +1,6 @@
 
 bandit
-numpy
+numpy<=2.0
```
Comment (achaikou) [inline python/requirements-dev.txt:3]: Why <= and not < ?

**Comment by achaikou** [comment]
If you manage to solve Bigendian issue, do force-push from the start and build is very long (over 1 hour). Maybe we'll get lucky.
Also maybe add the release commit. Remember that are 2 more docs places to update the version.

File: .github/workflows/wheels.yaml [hunk 4]
```diff
@@ -60,7 +60,7 @@ jobs:
             CMAKE_PREFIX_PATH="$CMAKE_PREFIX_PATH;C:/Program Files (x86)/layered-file-protocols/"
 
           CIBW_ARCHS: ${{ matrix.arch }}
-          CIBW_SKIP: pp* *-musllinux_* cp31*-win32
+          CIBW_SKIP: pp* *-musllinux_* cp31*-win32 cp312-*
```
[the remark this answers is not in the document]
Comment (ErlendHaa) [reply .github/workflows/wheels.yaml:65]: Numpy stopped with win32 wheels. And without numpy, dlisio cannot work

File: .github/workflows/wheels.yaml [hunk 5]
```diff
@@ -75,7 +77,7 @@ jobs:
     runs-on: ubuntu-latest
     if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
       - uses: actions/download-artifact@v2
```
Comment (achaikou) [inline .github/workflows/wheels.yaml:82]: To v3. 
It didn't complain because we didn't run it.

**Review by achaikou — APPROVED** [review]

**Pull request merged**